app/src/test/java/org/simple/clinic/sync/DataSyncTest.kt

```diff
@@ -47,20 +52,22 @@ class DataSyncTest {
   @Test
   fun `when syncing everything, if any of the syncs throws an error, the other syncs must not be affected`() {
     val modelSync1 = mock<ModelSync>()
-    val modelSync2 = mock<ModelSync>()
-    val modelSync3 = mock<ModelSync>()
-
     val (sync1Completable, sync1Consumer) = Completable.complete().subscriptionTest()
     whenever(modelSync1.syncConfig()).thenReturn(createSyncConfig(SyncGroup.DAILY))
     whenever(modelSync1.sync()).thenReturn(sync1Completable)
+    whenever(modelSync1.name).thenReturn("sync1")
 
+    val modelSync2 = mock<ModelSync>()
     val sync2Completable = Completable.error(RuntimeException("TEST"))
     whenever(modelSync2.syncConfig()).thenReturn(createSyncConfig(SyncGroup.DAILY))
     whenever(modelSync2.sync()).thenReturn(sync2Completable)
+    whenever(modelSync2.name).thenReturn("sync2")
```

Comment:
Is there a reason why we are returning the name for the test in this commit? Should we do that for other tests in this file as well? 🤔

Reply:
This was done because in the event of a failure, we push the sync name and sync events to analytics. The test was throwing an error in the Rx chain because the sync name was returning `null`.

Reply:
The rest of the tests do not verify failures (iirc), so they do not attempt to read the `name` property.